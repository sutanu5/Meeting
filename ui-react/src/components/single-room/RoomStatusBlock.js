import React from 'react';
import PropTypes from 'prop-types';

const Details = ({room, details}) => (
  <div id="single-room__details">
    { details.appointmentExists &&
      <div id="single-room__meeting-title">
        <span id="single-room__next-up">
          {details.nextUp}
        </span>
        <span id="single-room__meeting-subject">
          {room.Appointments[0].Subject}
        </span>
      </div>
    }
  </div>
);

const Time = ({room, details}) => (
  <div id="single-room__meeting-time">
    { details.appointmentExists &&
      new Date(parseInt(room.Appointments[0].Start, 10)).toLocaleTimeString([], {weekday: 'short', hour: '2-digit', minute: '2-digit'})
      + ' - ' + 
      new Date(parseInt(room.Appointments[0].End, 10)).toLocaleTimeString([], {hour: '2-digit', minute: '2-digit'})
    }
  </div>
);

const Organizer = ({room, details}) => {
  return(
    <div id="single-room__meeting-organizer">
      {details.appointmentExists &&
        room.Appointments[0].Organizer
      }
    </div>
  );
};

const RoomStatusBlock = ({ config, details, room }) => (
  <div className={room.Busy ? 'columns small-8 left-col busy' : 'columns small-8 left-col open'}>
    <div id="single-room__room-name">{room.Name}</div>
    <div id="single-room__room-status">{room.Busy ? config.statusBusy : config.statusAvailable}</div>

    <Details room={room} details={details} />
    <Time room={room} details={details} />
    <Organizer room={room} details={details} />

  </div>
);

RoomStatusBlock.propTypes = {
  room: PropTypes.object.isRequired,
  details: PropTypes.object,
  config: PropTypes.object
}

export default RoomStatusBlock;
