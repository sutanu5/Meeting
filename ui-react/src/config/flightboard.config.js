module.exports = {
  'board' : {
    'nextUp' : 'Next Up',
    'statusAvailable' : 'Open',
    'statusBusy' : 'Busy',
    'statusError' : 'Error'
  },

  'navbar' : {
    'title' : 'Conference Room Availability',
  },

  'roomFilter' : {
    'filterTitle' : 'Locations',
    'filterAllTitle' : 'All Conference Rooms',
  },
};
