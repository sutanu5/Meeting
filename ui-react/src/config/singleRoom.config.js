module.exports = {
  'nextUp' : 'Next Up',
  'statusAvailable' : 'Open',
  'statusBusy' : 'Busy',
  'upcomingTitle' : 'Upcoming',
};
