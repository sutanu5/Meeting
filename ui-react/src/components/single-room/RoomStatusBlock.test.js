import RoomStatusBlock from './RoomStatusBlock';
import * as config from '../../config/singleRoom.config.js';

describe('RoomStatusBlock Component', () => {

  let room, details;
  
  beforeEach(() => {
    room = {
      'Roomlist':'Test Roomlist',
      'Name':'Test Room',
      'RoomAlias':'test-room',
      'Email':'email@email.com',
      'Appointments':[
         {
            'Subject':'Meeting Subject',
            'Organizer':'Meeting Organizer',
            'Start':1532966400000,
            'End':1533344400000
         }
      ],
      'Busy':true
    };

    details = {
      appointmentExists: true,
      nextUp: ''
    };

    // config = {
    //   statusBusy: 'Busy',
    //   statusAvailable: 'Open'
    // };
  });

  it('renders correctly', () => {
    const wrapper = shallow(<RoomStatusBlock room={room} details={details} config={config} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders room.Name', () => {
    const wrapper = shallow(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('#single-room__room-name').text();
    expect(div).toBe(room.Name);
  });

  // room.Status tests

  it('renders "Busy" if room.Busy is true', () => {
    const wrapper = shallow(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('#single-room__room-status').text();
    expect(div).toBe(config.statusBusy);
  });

  it('renders "busy" class when room.Busy is true', () => {
    const wrapper = shallow(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('.busy').exists();
    expect(div).toBeTruthy();    
  });

  it('renders "Open" if room.Busy is false', () => {
    room.Busy = false;
    const wrapper = shallow(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('#single-room__room-status').text();
    expect(div).toBe(config.statusAvailable);
  });

  it('renders "open" classes when room.Busy is true', () => {
    room.Busy = false;
    const wrapper = shallow(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('.open').exists();
    expect(div).toBeTruthy();    
  });

  // /room.Status tests
  // Details tests

  it('renders Details', () => {
    const wrapper = shallow(<RoomStatusBlock room={room} details={details} config={config} />);

    const component = wrapper.find('Details').exists();
    expect(component).toBeTruthy();
  });

  it('renders Details with text if appointment exists', () => {
    const wrapper = mount(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('#single-room__details').text();
    expect(div).not.toBe('');
  });

  it('renders Details blank if appointment does not exist', () => {
    details.appointmentExists = false;
    const wrapper = mount(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('#single-room__details').text();
    expect(div).toBe('');
  });

  it('renders details.nextUp texts', () => {
    details.nextUp = config.nextUp + ':';
    const wrapper = mount(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('#single-room__next-up').text();
    expect(div).toBe(config.nextUp + ':');
  });

  it('renders room.Appointments[0].Subject', () => {
    const wrapper = mount(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('#single-room__meeting-subject').text();
    expect(div).toBe(room.Appointments[0].Subject);
  });

  // /Details tests
  // Time tests

  it('renders Time', () => {
    const wrapper = shallow(<RoomStatusBlock room={room} details={details} config={config} />);

    const component = wrapper.find('Time').exists();
    expect(component).toBeTruthy();
  });

  it('renders Time with text if appointment exists', () => {
    const wrapper = mount(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('#single-room__meeting-time').text();
    expect(div).not.toBe('');
  });

  it('renders Time blank if appointment does not exist', () => {
    details.appointmentExists = false;
    const wrapper = mount(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('#single-room__meeting-time').text();
    expect(div).toBe('');
  })

  it('receives number from room.Appointments[0].Start', () => {
    const wrapper = shallow(<RoomStatusBlock room={room} details={details} config={config} />);

    const roomProps = wrapper.find('Time').props().room;
    expect(roomProps.Appointments[0].Start).toBeNumber(); 
  });

  it('receives number from room.Appointments[0].End', () => {
    const wrapper = shallow(<RoomStatusBlock room={room} details={details} config={config} />);

    const roomProps = wrapper.find('Time').props().room;
    expect(roomProps.Appointments[0].End).toBeNumber(); 
  });

  // /Time tests
  // Organizer tests

  it('renders Organizer', () => {
    const wrapper = shallow(<RoomStatusBlock room={room} details={details} config={config} />);

    const component = wrapper.find('Organizer').exists();
    expect(component).toBeTruthy();
  });

  it('renders Organizer with text if appointment exists', () => {
    const wrapper = mount(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('#single-room__meeting-organizer').text();
    expect(div).not.toBe('');
  });

  it('renders Organizer blank if appointment does not exist', () => {
    details.appointmentExists = false;
    const wrapper = mount(<RoomStatusBlock room={room} details={details} config={config} />);

    const div= wrapper.find('#single-room__meeting-organizer').text();
    expect(div).toBe('');
  });

  it('renders room.Appointments[0].Oragnizer', () => {
    const wrapper = mount(<RoomStatusBlock room={room} details={details} config={config} />);

    const div = wrapper.find('#single-room__meeting-organizer').text();
    expect(div).toBe(room.Appointments[0].Organizer);
  });

  // /Organizer tests
});
