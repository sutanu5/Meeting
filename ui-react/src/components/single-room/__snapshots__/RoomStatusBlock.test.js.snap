// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomStatusBlock Component renders correctly 1`] = `
<div
  className="columns small-8 left-col busy"
>
  <div
    id="single-room__room-name"
  >
    Test Room
  </div>
  <div
    id="single-room__room-status"
  >
    Busy
  </div>
  <Details
    details={
      Object {
        "appointmentExists": true,
        "nextUp": "",
      }
    }
    room={
      Object {
        "Appointments": Array [
          Object {
            "End": 1533344400000,
            "Organizer": "Meeting Organizer",
            "Start": 1532966400000,
            "Subject": "Meeting Subject",
          },
        ],
        "Busy": true,
        "Email": "email@email.com",
        "Name": "Test Room",
        "RoomAlias": "test-room",
        "Roomlist": "Test Roomlist",
      }
    }
  />
  <Time
    details={
      Object {
        "appointmentExists": true,
        "nextUp": "",
      }
    }
    room={
      Object {
        "Appointments": Array [
          Object {
            "End": 1533344400000,
            "Organizer": "Meeting Organizer",
            "Start": 1532966400000,
            "Subject": "Meeting Subject",
          },
        ],
        "Busy": true,
        "Email": "email@email.com",
        "Name": "Test Room",
        "RoomAlias": "test-room",
        "Roomlist": "Test Roomlist",
      }
    }
  />
  <Organizer
    details={
      Object {
        "appointmentExists": true,
        "nextUp": "",
      }
    }
    room={
      Object {
        "Appointments": Array [
          Object {
            "End": 1533344400000,
            "Organizer": "Meeting Organizer",
            "Start": 1532966400000,
            "Subject": "Meeting Subject",
          },
        ],
        "Busy": true,
        "Email": "email@email.com",
        "Name": "Test Room",
        "RoomAlias": "test-room",
        "Roomlist": "Test Roomlist",
      }
    }
  />
</div>
`;
